### PR TITLE
fix: Return original document title when leaving a dashboard

### LINKED ISSUE #16254: [dashboard] page title does not change back after navigating away from dashboard
### Expected results

When navigating away from a dashboard the page title change back to "Superset"

### Actual results

The page title is still the dashboard's title 
#### Screenshots

Screen Shot 2021-08-13 at 2 04 07 PM
Screen Shot 2021-08-13 at 2 04 19 PM

#### How to reproduce the bug

1. view a dashboard, eg "World Bank's Data"
2. notice title of the page/tab/document is the name of the dashboard, eg "World Bank's Data"
3. navigate away via the menu, eg Data > Databases
4. notice the title of the page/tab/document is still the title of the dashboard, eg "World Bank's Data"

### Environment

- superset version: `master`
- python version: `3.7.5`
- node.js version: `14`
- any feature flags active: None

### Additional context

Very minor issue, likely happened when we move the dashboard "app" into the SPA bundle.

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY
This PR replaces the document title back to the original one when leaving the dashboard. This is to make sure that the dashboard title does not remain in the browser tab while moving to different pages without reloading.

This is not a definitive solution but a quick fix for the problem reported in #16254. As we move toward eliminating page reloads, the need for a title management implementation arises. In fact, the titles are currently only managed server-side, with the Dashboard page being the only exception.

### BEFORE

https://user-images.githubusercontent.com/60598000/129903630-e99f92f6-733b-4566-82a9-b7f9ff09a715.mp4

### AFTER

https://user-images.githubusercontent.com/60598000/129903675-df40022d-e4d8-443a-82cf-8e198e1eb8f1.mp4

### TESTING INSTRUCTIONS
1. Go to the Dashboards' list
2. Open a Dashboard and observe the browser tab title
3. Go back to the Dashboards' list and make sure the browser tab title has changed

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #16254
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -42,6 +42,8 @@ const DashboardContainer = React.lazy(
     ),
 );
 
+const documentTitle = document.title;
```
Comment (kgabryje) [inline superset-frontend/src/dashboard/containers/DashboardPage.tsx:45]: Can we rename it to something like `originalDocumentTitle` or `prevDocumentTitle` to make the intention more clear?

**Review by kgabryje — APPROVED** [review]
Tested, works nicely! 1 small nit about naming, but not blocking

**Review by villebro — APPROVED** [review]
LGTM. I wonder if we should start using document titles more to make it easier to distinguish between tab contents?

**Comment by geido** [comment]
> LGTM. I wonder if we should start using document titles more to make it easier to distinguish between tab contents?

We should  definitely look into this

**Comment by daphn3k** [comment]
> LGTM. I wonder if we should start using document titles more to make it easier to distinguish between tab contents?

I know this is a while ago, but was this ever picked up/is planned? I couldn't find something from my search.

**Pull request merged**